cinema/views.py

```diff
@@ -34,25 +37,84 @@ class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    @staticmethod
+    def _params_to_ints(query_string):
+        return [int(str_id) for str_id in query_string.split(",")]
+
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
 
         if self.action == "retrieve":
             return MovieDetailSerializer
 
-        return MovieSerializer
+        return self.serializer_class
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        date = self.request.query_params.get("date")
+        if actors:
+            actors = self._params_to_ints(actors)
+            queryset = queryset.filter(id__in=actors)
```

Comment:
In the `get_queryset` method of `MovieViewSet`, the filter for actors seems incorrect. It should filter movies by actors, not by movie IDs. Consider using `queryset.filter(actors__id__in=actors)` to filter movies that have the specified actors.